internal/parse/parse.go

```diff
@@ -0,0 +1,358 @@
+package parse
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	internalUtil "github.com/containers/buildah/internal/util"
+	"github.com/containers/common/pkg/parse"
+	"github.com/containers/storage"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	selinux "github.com/opencontainers/selinux/go-selinux"
+	"github.com/pkg/errors"
+)
+
+const (
+	// TypeBind is the type for mounting host dir
+	TypeBind = "bind"
+	// TypeTmpfs is the type for mounting tmpfs
+	TypeTmpfs = "tmpfs"
+	// TypeCache is the type for mounting a common persistent cache from host
+	TypeCache = "cache"
+	// mount=type=cache must create a persistent directory on host so its available for all consecutive builds.
+	// Lifecycle of following directory will be inherited from how host machine treats temporary directory
+	BuildahCacheDir = "buildah-cache"
+)
+
+var (
+	errBadMntOption = errors.Errorf("invalid mount option")
+	optionArgError  = errors.Errorf("must provide an argument for option")
+	noDestError     = errors.Errorf("must set volume destination")
+)
+
+// GetBindMount parses a single bind mount entry from the --mount flag.
```

Comment:
Please describe what the `string` return value is, and what the caller's supposed to do with it.  It's not obvious.

Reply:
added.

Reply:
It looks like the caller needs to unmount the image with the returned name, sometimes.  If we changed things so that the string value was _only_ returned when we mounted an image, and the value was understood to be the name (or better, the ID) of an image that needed to be unmounted when cleaning up, it would be much easier for callers to clean up after themselves.